src/main/scala/nl.knaw.dans.dccd.convert/DccdConvertExportApp.scala

```diff
@@ -15,10 +15,216 @@
  */
 package nl.knaw.dans.dccd.convert
 
-import scala.util.{ Try, Success }
+import java.io.{ File, FileWriter }
+import java.nio.file.{ Path, Paths }
+import java.text.SimpleDateFormat
+import java.util.{ Calendar, Date, GregorianCalendar, Locale }
 
-class DccdConvertExportApp(wiring: ApplicationWiring)  {
+import org.apache.commons.csv.{ CSVFormat, CSVPrinter }
 
-  // Application logic
+import scala.io.Source.fromString
+import scala.util.Try
+import scala.xml.parsing.ConstructingParser.fromSource
+import scala.xml.{ Elem, NodeSeq, TopScope, XML }
 
-}
\ No newline at end of file
+//class DccdConvertExportApp(wiring: ApplicationWiring)   {
+class DccdConvertExportApp(configuration: Configuration) {
+
+  def getListOfSubDirectories(directoryName: String): Array[String] = {
+    new File(directoryName)
+      .listFiles
+      .filter(_.isDirectory)
+      .map(_.getName)
+  }
+
+  def getListOfFiles(dir: String): List[File] = {
+    val d = new File(dir)
+    if (d.exists && d.isDirectory) {
+      d.listFiles.filter(_.isFile).toList
+    }
+    else {
+      List[File]()
+    }
+  }
+
+  def getPath(dir: String): Path = {
+    Paths.get(configuration.properties.getString(dir))
+  }
+
+  val csvFormat: CSVFormat = CSVFormat.RFC4180.withHeader("DATASET", "DC_TITLE", "DEPOSITOR_ID", "DCX_CREATOR_ORGANIZATION", "DDM_CREATED", "DCT_RIGHTSHOLDER", "DC_DESCRIPTION", "DC_SUBJECT", "DCT_TEMPORAL", "DCX_SPATIAL_X", "DCX_SPATIAL_Y", "DC_IDENTIFIER", "DCX_RELATION_QUALIFIER", "DCX_RELATION_TITLE", "DCX_RELATION_LINK", "DCX_RELATION_QUALIFIER(2)", "DCX_RELATION_TITLE(2)", "DC_TYPE", "DC_LANGUAGE", "DDM_ACCESSRIGHTS", "DDM_AVAILABLE", "DDM_AUDIENCE").withDelimiter(',')
+  val out: Appendable = new StringBuffer()
```

Comment:
Looks like you 'write' to a stringbuf and then, at the end, write the whole string to a file. Could you try to write to an output stream (from an instructions.csv file you create).